DOCSP-15810: update changestream page to use new feature + cleanup (#407)
mongodb/docs-java: source/fundamentals/crud/read-operations/change-streams.txt

   :start-after: begin aggregationExample
   :end-before: end aggregationExample

> An update operation on the collection should produce output similar to the
> following text:

Reviewer: I: [Style Guide on "should"](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#s:~:text=the%20setup%20procedure.-,should,-Use%20should%20only)

S: The text you deleted looks fine to me. Otherwise:

```suggestion
An update operation on the collection produces the following output:
```

Author: changed everywhere

You can configure the change event to contain or omit the following data:

> - The **pre-image**, which is a document that represents the version of the
>   document before the operation, if it exists
> - The **post-image**, which is a document that represents the version of the

Reviewer: S: Sounds lighter and cleaner to omit "which is"

```suggestion
- The **pre-image**, a document that represents the version of the
  document before the operation, if it exists
- The **post-image**, a document that represents the version of the
```

  document after the operation, if it exists

To receive change stream events that include a pre-image or post-image, you

> must connect to a deployment running MongoDB v6.0 or later and perform the
> following actions:

Reviewer: S: This sentence is a little hard to parse. Could we move the version info to an admonition or something?

```suggestion
must perform perform the following actions on your MongoDB deployment:
```

  .. tip::

     To configure your change stream to record the pre-image in change
     events, see the :ref:`<java-pre-image-example>`.

     To configure your change stream to record the post-image in change
     events, see the :ref:`<java-post-image-example>`.

.. _java-change-stream-pre-post-collection:

Create a Collection with Pre-Image and Post-Images Enabled
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To use the driver to create a collection that has the pre-image and post-image options

Reviewer: S: 
```suggestion
To use the driver to create a collection with the pre-image and post-image options
```

You can change the pre-image and post-image option in an existing collection
by running the ``collMod`` command from the MongoDB Shell. To learn how to
perform this operation, see the entry on :manual:`collMod </reference/command/collMod/>`
in the Server manual.

.. warning::

   When you modify this option on a collection, any change streams open on
   that collection in your application may fail if you configure them to require

>    the pre-image or post-image.

Reviewer: S: Might be able to trim this down a bit. [Style Guide](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#s:~:text=than%20matrices.-,may,-Use%20may%20to) also recommends using "may" only for permission

Maybe something like: 
"If you enable this option while change streams are open in your application, 
those change streams might fail."

Author: reworded

   :end-before: end fullDocumentBeforeChangeExample

The preceding example configures the change stream to use the
``FullDocumentBeforeChange.REQUIRED`` option. This option configures the change
stream to require pre-images for replace, update, and delete change
events. If the pre-image is not available, the driver raises an error.

Suppose you update the ``amount`` field of a document from ``150`` to

> ``2000``. The change event produced by this change should appear in the

Reviewer: ```suggestion
``2000``. The change event produced by this change appears in the
```

Author: changed along with changes from an earlier comment

   :end-before: end fullDocumentExample

The preceding example configures the change stream to use the

> ``FullDocument.WHEN_AVAILABLE`` option. This option configures the change
> stream to the change stream to return the post-image of the modified
> document for replace and update change events, if it is available.

Reviewer: ```suggestion
``FullDocument.WHEN_AVAILABLE`` option. This option configures the change
stream to return the post-image of the modified
document for replace and update change events, if it's available.
```

> Suppose you update the ``color`` field of a document from ``"purple"`` to

Reviewer: S: Could say "value of the color field," but this reads fine to me.
